rdflib/collection.py

```diff
@@ -231,6 +235,9 @@ def append(self, item: Node) -> Collection:
 
         """
 
+        if self.uri == RDF.nil:
+            raise ValueError("Cannot append to empty list")
+
         end = self._end()
```

Comment:
Can we make the same change here to `append()` too please?
```suggestion
        end = self._end()
        if end == RDF.nil:
            raise ValueError("Cannot append to empty list")

```

Reply:
Sorry, I missed it